docs/how-to/partitions.rst

```diff
@@ -0,0 +1,123 @@
+******************************************
+Adding partition support to an application
+******************************************
+
+Partitions basics
+=================
+
+In the below examples, we work with three partitions: ``default``, ``kernel``, and ``bar-baz``.  The ``default`` and ``kernel`` partitions do not have a namespace.  The ``bar-baz`` partition is part of the ``component`` namespace.
+
+``default`` must always be the first listed partition.
+
+Partition names and namespace names must consist of *only* lower-case alphabetic characters, unless a partition exists under a namespace, in which case it may also contain hyphen characters, though the first and last characters must still be alphabetic.
+
+.. _app_changes:
+
+Required application changes
+============================
+
+To add partition support to an application, two basic changes are needed to the ``craft-application``.
```

Comment:
This feels like a "How to enable partitions through craft-application` rather than a `craft-parts` how to.  I think your how-to from a craft-application perspective is more useful but perhaps this belongs in craft-application?

Ashort reference in craft-parts could cover the specifics like naming conventions, env vars, etc.

Reply:
Yeah that makes sense, I'll split it up.